lib/ecto/changeset.ex

```diff
@@ -2054,7 +2054,7 @@ defmodule Ecto.Changeset do
     constraint = opts[:name] || "#{get_source(changeset)}_#{get_field_source(changeset, field)}_index"
     message    = message(opts, "has already been taken")
     match_type = Keyword.get(opts, :match, :exact)
-    add_constraint(changeset, :unique, to_string(constraint), match_type, field, {message, []})
+    add_constraint(changeset, :unique, to_string(constraint), match_type, field, {message, [validation: :unique]})
```

Comment:
@josevalim just add an extra field of `kind` like the other ones.

Yes I can do that code but the point is to have consistency in how we do it. Extra information IMO is better, no?